ksqldb-engine/src/main/java/io/confluent/ksql/internal/ScalablePushQueryExecutorMetrics.java

```diff
@@ -0,0 +1,728 @@
+import io.confluent.ksql.util.KsqlConstants;
+import io.confluent.ksql.util.KsqlConstants.QuerySourceType;
+import io.confluent.ksql.util.KsqlConstants.RoutingNodeType;
+import io.confluent.ksql.util.ReservedInternalTopics;
+import java.io.Closeable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+import org.apache.kafka.common.metrics.MeasurableStat;
+import org.apache.kafka.common.metrics.Metrics;
+import org.apache.kafka.common.metrics.Sensor;
+import org.apache.kafka.common.metrics.stats.Avg;
+import org.apache.kafka.common.metrics.stats.CumulativeCount;
+import org.apache.kafka.common.metrics.stats.CumulativeSum;
+import org.apache.kafka.common.metrics.stats.Max;
+import org.apache.kafka.common.metrics.stats.Min;
+import org.apache.kafka.common.metrics.stats.Percentile;
+import org.apache.kafka.common.metrics.stats.Percentiles;
+import org.apache.kafka.common.metrics.stats.Percentiles.BucketSizing;
+import org.apache.kafka.common.metrics.stats.Rate;
+import org.apache.kafka.common.utils.Time;
+
+@SuppressWarnings("checkstyle:ClassDataAbstractionCoupling")
+public class ScalablePushQueryExecutorMetrics implements Closeable {
+
+  private static final String SCALABLE_PUSH_QUERY_METRIC_GROUP = "scalable-push-query";
+  private static final String SCALABLE_PUSH_REQUESTS = "scalable-push-query-requests";
+  private static final long MAX_LATENCY_BUCKET_VALUE_MICROS = TimeUnit.SECONDS.toMicros(10);
+  private static final int NUM_LATENCY_BUCKETS = 1000;
+
+  private final List<Sensor> sensors;
+  private final Sensor localRequestsSensor;
+  private final Sensor remoteRequestsSensor;
+  private final Sensor connectionDurationSensor;
+  private final Map<MetricsKey, Sensor> connectionDurationSensorMap;
+  private final Sensor errorRateSensor;
+  private final Map<MetricsKey, Sensor> errorRateSensorMap;
+  private final Sensor requestSizeSensor;
+  private final Sensor responseSizeSensor;
+  private final Map<MetricsKey, Sensor> responseSizeSensorMap;
+  private final Sensor responseCode2XX;
+  private final Sensor responseCode3XX;
+  private final Sensor responseCode4XX;
+  private final Sensor responseCode5XX;
+  private final Map<MetricsKey, Sensor> rowsReturnedSensorMap;
+  private final Map<MetricsKey, Sensor> rowsProcessedSensorMap;
+  private final Metrics metrics;
+  private final Map<String, String> legacyCustomMetricsTags;
+  private final Map<String, String> customMetricsTags;
+  private final String ksqlServiceIdLegacyPrefix;
+  private final String ksqlServicePrefix;
+  private final Time time;
+
+  public ScalablePushQueryExecutorMetrics(
+      final String ksqlServiceId,
+      final Map<String, String> customMetricsTags,
+      final Time time
+  ) {
+    this.ksqlServiceIdLegacyPrefix = ReservedInternalTopics.KSQL_INTERNAL_TOPIC_PREFIX
```

Comment:
We should be able to skip all of the legacy metrics since nothing is counting on them existing, so we can go right to the new method for metrics.